src/edu/csus/ecs/pc2/clics/API202306/CLICSOrganization.java

```diff
@@ -0,0 +1,167 @@
+
+    public CLICSOrganization() {
+        // for Jackson deserializer
+    }
+
+    /**
+     * Fill in organization information properties
+     * TODO: fix to use organization obtained from organizations.json.  From insitutions.tsv, we get this:
+     * [0]:1329 [1]:New York University [2]:NYU
+     * but only need [1].  We get the rest from the a typical account.
+     *
+     * @param account typical account using this organization
+     * @param orgFields String array of fields that correspond to institutions.tsv.  Gack!
+     */
+    public CLICSOrganization(String orgId, Account account, String [] orgFields) {
+        id = orgId;
+        icpc_id = orgId;
+        name = orgFields[2];
+        formal_name = orgFields[1];
+        country = account.getCountryCode();
+        if(country.equals(Constants.DEFAULT_COUNTRY_CODE)) {
+            country = null;
+        }
+    }
+
+    public String toJSON() {
+
+        try {
+            ObjectMapper mapper = JSONUtilities.getObjectMapper();
+            return mapper.writeValueAsString(this);
+        } catch (Exception e) {
+            return "Error creating JSON for organization info " + e.getMessage();
+        }
+    }
+
+    /**
+     * Create Organization (institution) map from a organizations.json like file
+     *
+     * @param jsonfile json file to deserialize
+     * @return Hashmap that maps institution id to an array[3] of Strings: id/code,formal name,name or null on error
+     */
+    public static HashMap<String, String[]> fromJSON(File jsonfile) {
+        HashMap<String, String[]>institutionsMap = new HashMap<String, String[]>();
+        Log log = StaticLog.getLog();
+        boolean error = false;
+
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            institutionsMap = createInstitutionsFromJSON(mapper.readValue(jsonfile, CLICSOrganization[].class), log);
+        } catch (Exception e) {
+            // deserialize exceptions
+            log.log(Log.WARNING, "could not deserialize organizations file " + jsonfile.toString(), e);
+        }
+        return(institutionsMap);
+     }
+
+    /**
+     * Create Organization (institution) map from a json string
+     *
+     * @param json string to deserialize
+     * @return Hashmap that maps institution id to an array[3] of Strings: id/code,formal name,name or null on error
+     */
+    public static HashMap<String, String[]> fromJSON(String json) {
+        HashMap<String, String[]>institutionsMap = null;
+        Log log = StaticLog.getLog();
+
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            institutionsMap = createInstitutionsFromJSON(mapper.readValue(json, CLICSOrganization[].class), log);
+            // deserialize exceptions
+        } catch (Exception e) {
+            log.log(Log.WARNING, "could not deserialize organizations string", e);
+        }
+        return(institutionsMap);
+    }
+
+    /**
+     * Convert CLICS organizations to PC2 hashma
```

Comment:
What's a "hashma"  :)

Reply:
Don't tell me you've never heard of a `hashma `?  It's the "mother" of all Hash Maps!  (I changed it to the lesser, `hashmap`, in the comment for the next push).